### ISSUE TITLE
Define API using OpenAPI standard

### ISSUE THREAD

**Issue by jlosito** [clapfoot/warapi#123]
Rather than defining this API using a README, I much rather view it via the OpenAPI standard which would provide much more features compared to what Clapfoot currently has. In addition, the file that would define the API could be used to host public documentation using GitHub pages.

https://spec.openapis.org/oas/latest.html#openapi-specification

**Comment by 444B** [comment]
I am in support of this and am happy to help implement it

**Comment by hayden-t** [comment]
nice idea, but it will never happen. 1- devs cant even fix bugs in api let alone completely restructure the format, 2 - it would break all existing integrations...

**Comment by jlosito** [comment]
@hayden-t can you expand on why it would break existing integrations?

**Comment by hayden-t** [comment]
because the data structure would be different to what existing integrations have been built to use

**Comment by jlosito** [comment]
@444B i just took a double look at the current schemas. I believe their api could be defined using the standard without breaking any integrations. Would you agree?

**Comment by hayden-t** [comment]
no, not unless the ran parallel output formats, any difference would be breaking, not to say integrations could adapt, but i can confidently say judging by progress here over the last 5+ years it aint going to happen.

**Comment by attrib** [comment]
Not sure if he just means something like this: https://attrib.github.io/warapi/

See https://github.com/attrib/warapi/blob/master/openapi.yaml

**Comment by jlosito** [comment]
@attrib it does! Thank you.
